NetCoreRestApi/DataLayer/Repositories/CategoryRepository.cs

```diff
@@ -4,32 +4,29 @@
 
 namespace DataLayer.Repositories
 {
-    public class CategoryRepository : IRepository<CategoryModel>
+    public class CategoryRepository<TId> : ICategoryRepository<TId>
     {
-        public CategoryModel GetById(int id)
+        public CategoryModel<TId> GetById(TId id)
         {
             throw new System.NotImplementedException();
         }
 
-        public IQueryable<CategoryModel> GetAll()
+        public IQueryable<CategoryModel<TId>> GetAll()
```

Comment:
Move all generic implementation to BaseRepository class

Reply:
done